config/wdio/visualRegressionConf.js

```diff
@@ -0,0 +1,86 @@
+const path = require('path');
+const LocalCompare = require('wdio-visual-regression-service/compare').LocalCompare;
+const viewports = require('./services.default-config').terraViewports;
+
+const screenshotSetup = {
+  diffDir: 'diff',
+  referenceDir: 'reference',
+  screenshotDir: 'latest',
+};
+
+const testIdRegex = /\[([^)]+)\]/;
+
+function createTestName(fullName) {
+  const matches = testIdRegex.exec(fullName);
+
+  // If test ID is provided, use the ID for a shorter test name, otherwise use the full name
+  let name = matches ? matches[1] : fullName.trim();
+
+  // Remove white space
+  name = name.replace(/[\s+.]/g, '_');
+
+  // Remove windows reserved characters. See: https://msdn.microsoft.com/en-us/library/windows/desktop/aa365247(v=vs.85).aspx#naming_conventions
+  // eslint-disable-next-line no-useless-escape
+  name = name.replace(/[\?\<\>\/\\|\*\"\:\+\.]/g, '-');
+
+  return name;
+}
+
+function getScreenshotName(context) {
+  const parentName = createTestName(context.test.parent);
+  const testName = createTestName(context.test.title);
+
+  return `${parentName}[${testName}].png`;
+}
+
+function getFormFactor(context) {
+  let formFactor = global.browser.options.formFactor;
+
+  if (!formFactor) {
+    const browserWidth = context.meta.viewport.width;
+    const viewportSizes = Object.keys(viewports);
+    for (let i = 0; i < viewportSizes.length; i += 1) {
+      const viewport = viewportSizes[i];
+      if (browserWidth <= viewports[viewport].width) {
+        formFactor = viewports[viewport].name;
+        break;
+      }
+    }
+  }
+
+  return formFactor;
+}
+
+function getScreenshotPath(ref, context) {
```

Comment:
I don't think `ref` is a very descriptive variable name in these couple of methods. Maybe `screenshotType` or `screenshotCategory` would be better.